### PR TITLE
Context forms

### PR DESCRIPTION
**Pull request by zrpnr**
The Reply and Edit forms needed the default text format #38 
That was introduced in #36 

Rather than pass the prop through several layers of parent components, I refactored some parts of this app to use the Context API. When props are not used directly in the rendered component it's more convenient to use the "hook" style for Context, so I updated the form component itself to use a new custom hook and also inherit the format.

The `isRefreshing` prop was getting passed around a lot so I added it to the same context.

### CODE REVIEW

File: js/src/useFluidForm.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { useState } from 'react';
+
+const useFluidForm = ({ subject = '', body = ''}) => {
```
Comment (gabesullice) [inline js/src/useFluidForm.js]: Maybe this should just be defined in `FluidCommentForm` until it has a use in > 1 place.

File: js/src/FluidCommentLink.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+import React, { useContext } from 'react';
 
-import React from 'react';
+import RequestContext from './RequestContext';
 
-const FluidCommentLink = ({ link, handleClick }) => (
-  <a href="#" onClick={handleClick}>{link.title}</a>
-);
+const FluidCommentLink = ({ text, handler }) => {
+  const request = useContext(RequestContext);
```
Comment (gabesullice) [inline js/src/FluidCommentLink.js]: Maybe `request` needs a name change too.

File: js/src/RequestContext.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React, { createContext } from 'react';
+
+const RequestContext = createContext({
```
[the remark this answers is not in the document]
Comment (zrpnr) [reply js/src/RequestContext.js]: sounds good

File: js/src/FluidCommentLink.js (hunk 2 above)
Comment (zrpnr) [reply js/src/FluidCommentLink.js]: can just destructure right away instead: `const { isRefreshing } = useContext(FluidCommentContext)`

File: js/src/RequestContext.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (zrpnr) [reply js/src/RequestContext.js]: Renamed RequestContext to FluidCommentContext

File: js/src/useFluidForm.js (hunk 1 above)
Comment (zrpnr) [reply js/src/useFluidForm.js]: Removed custom hook and added `useState` and the setter method directly into the form, also improved how the default values are set (this only applies to the "edit" form)

**Pull request merged**